src/earn/EarnEnterAmount.tsx

```diff
@@ -0,0 +1,628 @@
+            <View style={styles.inputRow}>
+              <AmountInput
+                inputRef={tokenAmountInputRef}
+                inputValue={tokenAmountInput}
+                onInputChange={onTokenAmountInputChange}
+                inputStyle={styles.inputText}
+                autoFocus
+                placeholder={new BigNumber(0).toFormat(2)}
+                testID="SendEnterAmount/TokenAmountInput"
+              />
+              <View style={styles.tokenView} testID="SendEnterAmount/TokenSelect">
+                <>
+                  <TokenIcon token={token} size={IconSize.SMALL} />
+                  <Text style={styles.tokenName}>{token.symbol}</Text>
+                </>
+              </View>
+            </View>
+            <View style={styles.localAmountRow}>
+              <AmountInput
+                inputValue={token.priceUsd ? localAmountInput : '-'}
+                onInputChange={onLocalAmountInputChange}
+                inputRef={localAmountInputRef}
+                inputStyle={styles.localAmount}
+                placeholder={`${localCurrencySymbol}${new BigNumber(0).toFormat(2)}`}
+                testID="SendEnterAmount/LocalAmountInput"
+                editable={!!token.priceUsd}
+              />
+              {!token.balance.isZero() && (
+                <Touchable
+                  borderRadius={MAX_BORDER_RADIUS}
+                  onPress={onMaxAmountPress}
+                  style={styles.maxTouchable}
+                  testID="SendEnterAmount/Max"
+                >
+                  <Text style={styles.maxText}>{t('max')}</Text>
+                </Touchable>
+              )}
+            </View>
+          </View>
+        </View>
+        {showNotEnoughBalanceForGasWarning && (
+          <InLineNotification
+            variant={NotificationVariant.Warning}
+            title={t('sendEnterAmountScreen.notEnoughBalanceForGasWarning.title', {
+              feeTokenSymbol: prepareTransactionsResult.feeCurrencies[0].symbol,
+            })}
+            description={t('sendEnterAmountScreen.notEnoughBalanceForGasWarning.description', {
+              feeTokenSymbol: prepareTransactionsResult.feeCurrencies[0].symbol,
+            })}
+            style={styles.warning}
+            testID="SendEnterAmount/NotEnoughForGasWarning"
+          />
+        )}
+        {prepareTransactionError && (
+          <InLineNotification
+            variant={NotificationVariant.Error}
+            title={t('sendEnterAmountScreen.prepareTransactionError.title')}
+            description={t('sendEnterAmountScreen.prepareTransactionError.description')}
+            style={styles.warning}
+            testID="SendEnterAmount/PrepareTransactionError"
+          />
+        )}
+        <EarnProceed
+          tokenAmount={tokenAmount}
+          localAmount={localAmount}
+          token={token}
+          amountEnteredIn={enteredIn}
+          onPressProceed={onPressContinue}
+          onPressInfo={onPressInfo}
```

Comment:
Still need to pass this here. I tried having EarnProceed be a const inside of EarnEnterAmount rather than a function outside of it, but this was causing it to re-render everytime the input amount was changed and so the APY was flashing a lot.